core/controller/src/main/scala/org/apache/openwhisk/core/controller/Packages.scala

```diff
@@ -299,9 +299,14 @@ trait WhiskPackagesApi extends WhiskCollectionAPI with ReferencedEntities {
       case b: Binding =>
         val docid = b.fullyQualifiedName.toDocId
         logging.debug(this, s"fetching package '$docid' for reference")
-        getEntity(WhiskPackage.get(entityStore, docid), Some {
-          mergePackageWithBinding(Some { wp }) _
-        })
+        if (docid == wp.docid) {
+          logging.error(this, "unexpected package binding refers to itself: $docid")
+          terminate(InternalServerError)
```

Comment:
Two points about this:
1) I'm not quite sure I 100% agree with a `400 Bad Request` in this situation as there is nothing in fact wrong with the request, however there is entirely something wrong with the stored state of the server. While I also don't agree with a `500 Internal Server Error` I don't see a better alternative for either. While it's use is intended for WebDAV, it almost appears that `508 Loop Detected` would be the most accurate description :)
2) There is the attempt at performing the `get` on the object afterward, and that the test checks it is a 200 response, however you're correct that with the fix for ingest in place it would only be possible to manually craft this bad record (and I'm unable to determine how to do that given the current tests)

Reply:
@rabbah WDYT?

Reply:
Seems like a 422 is pretty spot on "422 UNPROCESSABLE ENTITY"

https://tools.ietf.org/html/rfc4918#section-11.2

Reply:
I can agree on `422 Unprocessable Entity` - I'll make those changes.